### PR TITLE
[MIGR]: product_email_template: Migrated into new api.

### PR DESCRIPTION
**Pull request by ppa-odoo**
**Task**: https://www.odoo.com/web#id=12150&view_type=form&model=project.task&menu_id=3942&action=327
**Pad**: https://pad.odoo.com/p/r.1uVHqWtuk9eCbOQ9

### CODE REVIEW

File: addons/product_email_template/__openerp__.py [hunk 1]
```diff
@@ -15,8 +15,8 @@
         'data/product_demo.xml',
     ],
     'data': [
-        'views/product_view.xml',
-        'views/email_template_view.xml',
+        'views/product_views.xml',
+        'views/mail_template_views.xml',
     ],
     'installable': True,
     'auto_install': False,
```
Comment (atp-odoo) [inline addons/product_email_template/__openerp__.py]: default value of auto_install is False. no need to set. remove it

File: addons/product_email_template/models/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def invoice_validate_send_email(self):
+        for invoice in self:
+            # send template only on customer invoice
+            if invoice.type != 'out_invoice':
```
Comment (atp-odoo) [inline addons/product_email_template/models/account_invoice.py]: cannot set filter for this in self.filtered(lambda x: x.type != 'out_invoice')?

File: addons/product_email_template/models/account_invoice.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                if line.product_id.email_template_id:
+                    # CLEANME: should define and use a clean API: message_post with a template
+                    Composer = self.env['mail.compose.message'].create({
+                        'model': 'account.invoice',
+                        'res_id': invoice.id,
+                        'template_id': line.product_id.email_template_id.id,
+                        'composition_mode': 'comment',
+                    })
+                    template_values = Composer.onchange_template_id(line.product_id.email_template_id.id, 'comment', 'account.invoice', invoice.id)['value']
+                    template_values['attachment_ids'] = [(4, id) for id in template_values.get('attachment_ids', [])]
```
Comment (atp-odoo) [inline addons/product_email_template/models/account_invoice.py]: use proper name. like [ [(4, attach) for attach in template_values.get('attachment_ids')]]

File: addons/product_email_template/models/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+
+from openerp import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
```
Comment (atp-odoo) [inline addons/product_email_template/models/account_invoice.py]: useless decorator ;)

File: addons/product_email_template/views/mail_template_views.xml [hunk 5]
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
-    <data>
-        <record model="ir.ui.view" id="email_template_form_simplified">
+<odoo>
+        <record id="email_template_form_simplified" model="ir.ui.view">
             <field name="name">email.template.form.simplified</field>
```
Comment (atp-odoo) [inline addons/product_email_template/views/mail_template_views.xml]: now name should be "mail.template.form.simplified."

File: addons/product_email_template/models/account_invoice.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    def invoice_validate_send_email(self):
+        for invoice in self.filtered(lambda x: x.type != 'out_invoice'):
```
Comment (pna-odoo) [inline addons/product_email_template/models/account_invoice.py]: wrong domain passed, it should be like x.type != 'in_invoice'

File: addons/product_email_template/data/product_demo.xml [hunk 7]
```diff
@@ -66,35 +65,35 @@
         Each day of the week is dedicated to specific applications in Odoo as per the following planning:
         <ul style="color: #6d57e0">
             <li>
-                <strong><a href="https://www.openerp.com/online-training#Monday" style="text-decoration:none;" target=_new>Monday: </a></strong>
-                <a href="https://www.openerp.com/online-training#Monday_1" style="text-decoration:none;" target=_new>Introduction</a>,
-                <a href="https://www.openerp.com/online-training#Monday_2" style="text-decoration:none;" target=_new>CRM</a>,
-                <a href="https://www.openerp.com/online-training#Monday_3" style="text-decoration:none;" target=_new>Sales Management</a>
+                <strong><a href="https://www.odoo.com/online-training#Monday" style="text-decoration:none;" target=_new>Monday: </a></strong>
```
Comment (mart-e) [inline addons/product_email_template/data/product_demo.xml]: the links `https://www.odoo.com/online-training` do not work

File: addons/product_email_template/models/product.py [hunk 8]
```diff
@@ -1,19 +1,16 @@
...
 
-    _columns = {
-        'email_template_id': fields.many2one(
-            'mail.template', 'Product Email Template',
-            help='When validating an invoice, an email will be sent to the customer '
-                 'based on this template. The customer will receive an email for each '
-                 'product linked to an email template.'),
-    }
+    email_template_id = fields.Many2one('mail.template', string='Product Email Template',
+        help='When validating an invoice, an email will be sent to the customer'
```
Comment (mart-e) [inline addons/product_email_template/models/product.py]: @ppa-odoo you reintroduced an old bug in the tooltip fixed at 38b1485d525afcdd4e28ae8a40bb1c3db3e52bdf

**Pull request merged**